src/main/java/com/google/step/servlets/DistanceUtil.java

```diff
@@ -0,0 +1,139 @@
+package com.google.step.servlets;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+import com.google.step.model.Deal;
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.http.client.utils.URIBuilder;
+
+public class DistanceUtil {
+
+  private static Gson gson = new Gson();
+  private static final String FAKE_API_KEY = "api-key";
+  private static final String API_KEY = readMapApiKey();
+
+  // Class to represent the json object returned from distance matrix api
+  private class DistanceResponse {
+    private class Row {
+      private List<Element> elements;
+
+      private class Element {
+        private Distance distance;
+        private Duration duration;
+        private String status;
+
+        private class Distance {
+          String text;
+          String value;
+        }
+
+        private class Duration {
+          String text;
+          String value;
+        }
+      }
+    }
+
+    private String status;
+    private List<String> origin_addresses;
+    private List<String> destination_addresses;
+    private List<Row> rows;
+  }
+
+  public static List<Map<String, Double>> getDistances(
+      List<Deal> deals, String latitude, String longitude, List<List<String>> placeIdsPerDeal)
+      throws IOException {
+    if (API_KEY.equals("api-key")) {
+      return new ArrayList<>();
+    }
+    HttpRequestFactory requestFactory = new NetHttpTransport().createRequestFactory();
+    StringBuilder sb = new StringBuilder();
+
+    // Building url with all the placeids of all the deals
+    for (int i = 0; i < placeIdsPerDeal.size(); i++) {
+      List<String> placeIdsOfDeal = placeIdsPerDeal.get(i);
+      for (int j = 0; j < placeIdsOfDeal.size(); j++) {
+        sb.append("place_id:");
+        sb.append(placeIdsOfDeal.get(j));
+        if (j != placeIdsOfDeal.size() - 1 || i != placeIdsPerDeal.size() - 1) {
+          sb.append("|");
+        }
+      }
+    }
```

Comment:
There's a much nicer way of doing this -
```suggestion
    String destinations = placeIdsPerDeal.stream()
    	.flatMap(placeIdsForDeal -> placeIdsForDeal.stream())
    	.map(placeId -> "place_id:"+placeId)
    	.collect(Collectors.joining("|"));
```